app/pages/settings/email.jsx

```diff
@@ -0,0 +1,300 @@
+import React from 'react';
+import apiClient from 'panoptes-client/lib/api-client';
+import talkClient from 'panoptes-client/lib/talk-client';
+import AutoSave from '../../components/auto-save';
+import handleInputChange from '../../lib/handle-input-change';
+
+class EmailSettingsPage extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      page: 1,
+      projects: [],
+      projectPreferences: [],
+      talkPreferences: []
+    };
+    this.handlePreferenceChange = this.handlePreferenceChange.bind(this);
+    this.nameOfPreference = this.nameOfPreference.bind(this);
+    this.sortPreferences = this.sortPreferences.bind(this);
+    this.talkPreferenceOption = this.talkPreferenceOption.bind(this);
+    this.getProjectForPreferences(props.user);
+  }
+
+  componentDidMount() {
+    talkClient.type('subscription_preferences').get()
```

Comment:
Out of interest, why are you running the API request for project preferences in the constructor, but the API request for Talk preferences after the component mounts?

Reply:
Oh, that's just me using `componentDidMount()` as the [default place where to fetch data](https://daveceddia.com/where-fetch-data-componentwillmount-vs-componentdidmount/). Project prefs were explicitly fetched in `componentWillMount`, which can be effectively replaced by the constructor. There's probably not much in it, but I'm happy to either move this to the constructor or the talk preferences call inside here. Have you got a preference?

Reply:
No strong preference, and the tests don't seem to be affected by using `componentDidMount`, but both calls should probably go in the same place.